File: 수학(구현)/백준_1011_Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.cpp
#include <iostream>
#include <cmath>
using namespace std;


int main()
{
	int n;
	scanf("%d", &n);
	long long start, end;
	long long term = 0;
	while (n--) {
		scanf("%d %d", &start, &end);
		term = end - start;
		
		if (term == 1) printf("1\n");
		else if (term == 2) printf("2\n");
		else {
			long long num1 = 2;
			long long num_plus = 1;
			long long num2 = 3;
			long long cnt = 2;

			while (1) {
				cnt++;
				num1 += num_plus;
				num_plus++;
				num2 += num_plus;
				if (num1 <= term && term < num2) break;
				
				cnt++;
				num1 += num_plus;
				num2 += num_plus;
				if (num1 <= term && term < num2) break;
			}
			printf("%lld\n", cnt);
		}
	}
	

}

File: 수학(구현)/백준_1011_Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.cs
using System.Collections;
using static System.Console;
using System;

class Program
{
    
    public static void Main()
    {
        string[] s = ReadLine().Split(' ');
        int n = int.Parse(s[0]);
        long start, end;
        long term = 0;

        while(n-- > 0)
        {
            s = ReadLine().Split(' ');
            start = int.Parse(s[0]);
            end = int.Parse(s[1]);
            term = end - start;

            if (term == 1) WriteLine("1");
            else if (term == 2) WriteLine("2");
            else
            {
                long num1 = 2;
                long num_plus = 1;
                long num2 = 3;
                long cnt = 2;

                while (true)
                {
                    cnt++;
                    num1 += num_plus;
                    num_plus++;
                    num2 += num_plus;
                    if (num1 <= term && term < num2) break;

                    cnt++;
                    num1 += num_plus;
                    num2 += num_plus;
                    if (num1 <= term && term < num2) break;
                }
                WriteLine(cnt);
            }
        }
     
    }
}